src/api/libcellml/componententity.h

```diff
@@ -311,8 +313,10 @@ class LIBCELLML_EXPORT ComponentEntity: public NamedEntity
      * implement their own versions.
      *
      * @param component The ComponentPtr to add to the list of components.
+     *
+     * @return True if the component was added, false otherwise.
      */
-    virtual void doAddComponent(const ComponentPtr &component);
+    virtual bool doAddComponent(const ComponentPtr &component);
```

Comment:
I can't recall when `doAddComponent ()` was added to `ComponentEntity`, but it has nothing to do here and should be moved to `ComponentEntity`'s private implementation. (The same holds true with `doAddComponent()` in `Model` and `Component`.)

Reply:
`doAddComponent` is the private virtual function for `addComponent` this is done to allow us to add API without breaking the ABI.  So no these methods cannot go into the private implementation class (until there is a whole private inheritance structure in place).

Reply:
Cool, one more reason for doing things "right" when it comes to our private inheritance structure. :)